src/System.Security.Cryptography.Csp/tests/RSACryptoServiceProviderTests.cs

```diff
@@ -270,6 +277,118 @@ public static void NonExportable_Persisted()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public static void Ctor_UseCspParameter_Throws_Unix()
+        {
+            var cspParameters = new CspParameters();
+            Assert.Throws<PlatformNotSupportedException>(() => new RSACryptoServiceProvider(cspParameters));
+            Assert.Throws<PlatformNotSupportedException>(() => new RSACryptoServiceProvider(0, cspParameters));
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public static void CspKeyContainerInfo_Throws_Unix()
+        {
+            var rsa = new RSACryptoServiceProvider();
+            Assert.Throws<PlatformNotSupportedException>(() => (rsa.CspKeyContainerInfo));
+        }
+
+        [Fact]
+        public static void ImportParameters_ExponentTooBig_Throws()
+        {
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                // Verify that Unix shims and Windows Csp both throws the same exception when large Exponent imported
+                Assert.ThrowsAny<CryptographicException>(() => rsa.ImportParameters(TestData.RsaBigExponentParams));
+            }
+        }
+
+        [Fact]
+        public static void SignHash_DefaultAlgorithm_Success()
+        {
+            byte[] hashVal;
+            using (SHA1 sha1 = SHA1.Create())
+            {
+                hashVal = sha1.ComputeHash(TestData.HelloBytes);
+            }
+
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                byte[] signVal = rsa.SignHash(hashVal, null);
+                Assert.True(rsa.VerifyHash(hashVal, null, signVal));
+            }
+        }
+
+        [Fact]
+        public static void VerifyHash_DefaultAlgorithm_Success()
+        {
+            byte[] hashVal;
+            using (SHA1 sha1 = SHA1.Create())
+            {
+                hashVal = sha1.ComputeHash(TestData.HelloBytes);
+            }
+
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                byte[] signVal = rsa.SignData(TestData.HelloBytes, "SHA1");
+                Assert.True(rsa.VerifyHash(hashVal, null, signVal));
+            }
+        }
+
+        [Fact]
+        public static void Encrypt_InvalidPaddingMode_Throws()
+        {
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                Assert.Throws<CryptographicException>(() => rsa.Encrypt(TestData.HelloBytes, RSAEncryptionPadding.OaepSHA256));
```

Comment:
Similarly, sign/verify should fail with RSASignaturePadding.PSS (if we don't already have tests for that)